dietpi/dietpi-config

```diff
@@ -430,32 +430,76 @@ A long (or insufficiently manufactured) cable may required a higher boost settin
 
 		elif (( $G_WHIP_RETURNED_VALUE == 16 )); then
 
-			# Known FP's for setting brightness
-			local afp_current_set_brightness=(
-
-				'/sys/class/backlight/intel_backlight'
-				'/sys/class/backlight/rpi_backlight'
-				'/sys/class/backlight/lcd0'
-				'/sys/class/backlight/backlight'
-			)
-
-			MIN_VALUE=0
-			MAX_VALUE=255
-			local fp_brightness current_brightness
-			for fp_brightness in "${afp_current_set_brightness[@]}"
+			local aFp_Brightness_Available=()
+			local aBrightness_Name=()
+			local aBrightness_Current=()
+			local aBrightness_Max=()
+
+			local iTotalAvailable=0
+			local fp_brightness
+			for fp_brightness in /sys/class/backlight/*
 			do
-				[[ -f $fp_brightness/brightness ]] || continue
-				current_brightness=$(<"$fp_brightness/brightness")
-				[[ -f $fp_brightness/max_brightness ]] && MAX_VALUE=$(<"$fp_brightness/max_brightness")
-				break
+				if [[ -f $fp_brightness/brightness && -f $fp_brightness/max_brightness ]] &&
+					disable_error=1 G_CHECK_VALIDINT "$(<"$fp_brightness/brightness")" &&
+					disable_error=1 G_CHECK_VALIDINT "$(<"$fp_brightness/max_brightness")"
+				then
+					aFp_Brightness_Available+=( "$fp_brightness" )
+					aBrightness_Name+=( "${fp_brightness##*/}" )
+					aBrightness_Current+=( "$(<$fp_brightness/brightness)" )
+					aBrightness_Max+=( "$(<$fp_brightness/max_brightness)" )
```

Comment:
Indeed that's it, also command substitutions, as `<$fp_brightness/brightness` would be a syntax error if `$fp_brightness` contained a space and it if it did contain a special character like `[]` or `*` and by chance a matching file exists, it can be glob-expanded:
```suggestion
					aBrightness_Current+=( "$(<"$fp_brightness/brightness")" )
					aBrightness_Max+=( "$(<"$fp_brightness/max_brightness")" )
```